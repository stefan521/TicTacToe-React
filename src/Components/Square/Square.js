import React from 'react';

export default function Square(props) {
  return (
    <button className="square" onClick={props.onClick}
      style={{background: props.color}}>
      {props.value}
    </button>
  );
}
