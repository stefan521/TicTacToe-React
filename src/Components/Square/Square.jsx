// @flow
import React from 'react';

type Props = {
  onClick: Function,
  winSquare: boolean,
  value: string
};

export default function Square(props: Props) {
  const { onClick } = props;
  const { winSquare } = props;
  const { value } = props;

  return (
    <div className="grid-item">
      <button
        type="button"
        className={winSquare ? 'win-square square' : 'square'}
        onClick={onClick}
      >
        {value}
      </button>
    </div>
  );
}
