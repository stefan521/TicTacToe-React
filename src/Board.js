import React from 'react';
import Square from './Square.js';

class Board extends React.Component {
  renderSquare(squareNumber) {
    return (
       <Square
          value={this.props.squares[squareNumber]}
          onClick={() => this.props.onClick(squareNumber)}
          key={squareNumber}
       />
    );
  }

  render() {
    let rows = [];
    let boardSize = 3;

    for(let i = 0; i < boardSize; i++) {
      let row = [];
      for(let j = 0; j < boardSize; j++) {
        let squareNumber = i * boardSize + j;
        row.push(this.renderSquare(squareNumber));
      }
      rows.push(<div className="board-row" key={i}>{row}</div>);
    }

    return (<div>{rows}</div>);
  }
}

export default Board;
