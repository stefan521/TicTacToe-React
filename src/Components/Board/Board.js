import React from 'react';
import Square from '../Square/Square.js';

class Board extends React.Component {
  pickSquareColor(squareNumber) {
    let winLine = this.props.winLine;
    if(winLine && winLine.includes(squareNumber)){
      return "lightgreen";
    } else {
      return "white";
    }
  }

  renderSquare(squareNumber) {
    return (
       <Square
          value={this.props.squares[squareNumber]}
          onClick={() => this.props.onClick(squareNumber)}
          color={this.pickSquareColor(squareNumber)}
          key={squareNumber}
       />
    );
  }

  render() {
    let rows = [];
    let boardSize = 3;

    for(let i = 0; i < boardSize; i++) {
      let row = [];
      for(let j = 0; j < boardSize; j++) {
        let squareNumber = i * boardSize + j;
        row.push(this.renderSquare(squareNumber));
      }
      rows.push(<div className="board-row" key={i}>{row}</div>);
    }

    return (<div>{rows}</div>);
  }
}

export default Board;
