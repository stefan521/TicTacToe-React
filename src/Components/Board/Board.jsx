// @flow
import React from 'react';
import Square from '../Square/Square.jsx';

type Props = {
  winLine: Array<number>,
  squares: Array<string>,
  onClick: Function
};

type State = {
  boardSize: number;
};

class Board extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      boardSize: 9,
    };
  }

  isWinSquare(squareNumber: number) {
    const { winLine } = this.props;
    if (winLine && winLine.includes(squareNumber)) {
      return true;
    }

    return false;
  }

  renderSquare(squareNumber: number) {
    const { squares } = this.props;
    const { onClick } = this.props;

    return (
      <Square
        value={squares[squareNumber]}
        onClick={() => onClick(squareNumber)}
        winSquare={this.isWinSquare(squareNumber)}
        key={squareNumber}
      />
    );
  }

  render() {
    // eslint-disable-next-line prefer-const
    let squares = Array(9);
    const { boardSize } = this.state;

    for (let i = 0; i < boardSize; i += 1) {
      squares[i] = this.renderSquare(i);
    }

    return (
      <div className="board-wrapper">
        <div className="game-board">{squares}</div>
      </div>
    );
  }
}

export default Board;
