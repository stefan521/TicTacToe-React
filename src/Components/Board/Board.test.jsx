import React from 'react';
import { shallow } from 'enzyme';
import Board from './Board.jsx';
import Square from '../Square/Square.jsx';

describe('Board', () => {
  const clickHandler = jest.fn();

  it('contains 9 squares', () => {
    const wrapper = shallow(<Board
      squares={Array(9).fill('X')}
      onClick={clickHandler}
      winLine={[1, 2, 3]}
    />);

    expect(wrapper.find(Square)).toHaveLength(9);
    expect(wrapper).toMatchSnapshot();
  });
});
