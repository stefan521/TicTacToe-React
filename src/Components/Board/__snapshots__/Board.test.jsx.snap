// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board contains 9 squares 1`] = `
<div
  className="board-wrapper"
>
  <div
    className="game-board"
  >
    <Square
      key="0"
      onClick={[Function]}
      value="X"
      winSquare={false}
    />
    <Square
      key="1"
      onClick={[Function]}
      value="X"
      winSquare={true}
    />
    <Square
      key="2"
      onClick={[Function]}
      value="X"
      winSquare={true}
    />
    <Square
      key="3"
      onClick={[Function]}
      value="X"
      winSquare={true}
    />
    <Square
      key="4"
      onClick={[Function]}
      value="X"
      winSquare={false}
    />
    <Square
      key="5"
      onClick={[Function]}
      value="X"
      winSquare={false}
    />
    <Square
      key="6"
      onClick={[Function]}
      value="X"
      winSquare={false}
    />
    <Square
      key="7"
      onClick={[Function]}
      value="X"
      winSquare={false}
    />
    <Square
      key="8"
      onClick={[Function]}
      value="X"
      winSquare={false}
    />
  </div>
</div>
`;
